### PR TITLE
feat: evidence and artifact resource handling

### LINKED ISSUE #651: SAR: Observations and Evidence - Evidence- 4.4.3
**Extended Description**
- As an ASAP developer, in order to assist FedRAMP reviewers, I want to examine the Observation assembly in the Evidence case for the SAR document for adherence to the Guide instructions.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- [x] a back-matter resource with a type property of 'artifact' or 'evidence' must have a matching uuid value from the observation/relevant-evidence/@href value (following the '#') or from the observation/subject/@subject-uuid value.
- [x] A resource of type property 'artifact' or 'evidence' must have an rlink and/or a base64 element children.  Multiple are acceptable.
- [x] any rlink/@href must contain a relative path.

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%
 - [x] Automated code quality checks passed
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed – needs to be created/automated
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
Closes #651
evidence and artifact resource handling

### CODE REVIEW

File: src/validations/rules/sar.sch [hunk 1]
```diff
@@ -128,6 +128,46 @@
...
+                test="
+                    if (@uuid[. = $relevant-evidence-href])
+                    then
+                        true()
+                    else
+                        if (@uuid[. = $subject-UUID])
+                        then
+                            true()
+                        else
+                            false()">Resources of type 'evidence' or 'artifact' have a matching relevant-evidence or subject
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:153]:
```suggestion
                            false()">Resources of type 'evidence' or 'artifact' must have a matching relevant-evidence or subject
```

File: src/validations/rules/sar.sch [hunk 2]
```diff
@@ -128,6 +128,46 @@
...
+                UUID.</sch:assert>
+            <sch:assert
+                diagnostics="has-type-artifact-evidence-rlink-base64-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.4.4"
+                id="has-type-artifact-evidence-rlink-base64"
+                role="error"
+                test="oscal:rlink or oscal:base64">An observation of type 'evidence' or 'artifact' has at least one rlink and/or base64
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:160]:
```suggestion
                test="oscal:rlink or oscal:base64">An observation of type 'evidence' or 'artifact' must have at least one rlink and/or base64
```

File: src/validations/rules/sar.sch [hunk 3]
```diff
@@ -128,6 +128,46 @@
...
+                element.</sch:assert>
+            <sch:assert
+                diagnostics="has-type-artifact-evidence-rlink-relative-path-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.4.4"
+                id="has-type-artifact-evidence-rlink-relative-path"
+                role="error"
+                test="oscal:rlink[contains(@href, './')]">If an observation of type 'evidence' or 'artifact' has an rlink the @href value is a
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:167]:
```suggestion
                test="oscal:rlink[contains(@href, './')]">If an observation of type 'evidence' or 'artifact' has an rlink the @href value must be a
```

**Review by danielnaab — CHANGES_REQUESTED** [review]

File: src/validations/rules/sar.sch [hunk 4]
```diff
@@ -128,6 +128,46 @@
...
+                id="has-type-artifact-evidence-rlink-base64"
+                role="error"
+                test="oscal:rlink or oscal:base64">An observation of type 'evidence' or 'artifact' must have at least one rlink and/or base64
+                element.</sch:assert>
+            <sch:assert
+                diagnostics="has-type-artifact-evidence-rlink-relative-path-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.4.4"
+                id="has-type-artifact-evidence-rlink-relative-path"
+                role="error"
+                test="oscal:rlink[contains(@href, './')]">If an observation of type 'evidence' or 'artifact' has an rlink the @href value must be a
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:167]: Determining if a link is relative is probably best done by ensuring there is no leading `/`. A relative link could be simply a file/document name. Even relative links can begin with a URI scheme.

**Review by GaryGapinski — APPROVED** [review]

File: src/validations/rules/sar.sch (hunk 4 above)
Comment (markXLIX) [reply src/validations/rules/sar.sch]: Sounds good.  Adjusted as such.

**Review by danielnaab — APPROVED** [review]

**Pull request merged**